File: kit/android/ble-payment-kit/src/main/kotlin/ru/paymentguide/blepaymentkit/BlePacketRejectReason.kt
package ru.paymentguide.blepaymentkit

enum class BlePacketRejectReason {
    invalidPrefix,
    malformedPayload,
    missingRequiredField,
    weakRSSI,
    expiredPacket,
    unsupportedVersion,
    duplicatePacket,
    signalBelowThreshold,
    packetTooShort,
    invalidQrcId,
    unknown,
}

File: ios/BlePaymentKit/Sources/BlePaymentKit/BlePacketRejectReason.swift
import Foundation

public enum BlePacketRejectReason: String, Error, Codable, Equatable, Sendable {
    case invalidPrefix
    case malformedPayload
    case missingRequiredField
    case weakRSSI
    case expiredPacket
    case unsupportedVersion
    case duplicatePacket
    case signalBelowThreshold
    case packetTooShort
    case invalidQrcId
    case unknown
}
